gcpvault.go

```diff
@@ -433,46 +425,71 @@ func newJWTBase(ctx context.Context, cfg Config) (string, error) {
 			Base:   hc.Transport,
 		},
 	}
-	iamClient, err := iam.New(hcIAM)
+
+	claim, err := json.Marshal(map[string]interface{}{
+		"aud": "vault/" + cfg.Role,
+		"sub": serviceAccount,
+		"exp": time.Now().UTC().Add(5 * time.Minute).Unix(),
+	})
+
 	if err != nil {
-		return "", errors.Wrap(err, "unable to init IAM client")
+		return "", errors.Wrap(err, "unable to encode JWT payload")
 	}
 
+	gcpURL := "https://iamcredentials.googleapis.com/v1"
```

Comment:
could be a constant

Reply:
we are reassigning the value is case of tests, so I don't think we can make it a const.